flocker/control/test/test_httpapi.py

```diff
@@ -2666,3 +2666,118 @@ def test_deleted(self):
             expected,
             api_dataset_from_dataset_and_node(dataset, expected_hostname)
         )
+
+
+class ContainerStateTestsMixin(APITestsMixin):
+    """
+    Tests for the containers state endpoint at ``/state/containers``.
+    """
+    def test_empty(self):
+        """
+        When the cluster state includes no containers, the endpoint
+        returns an empty list.
+        """
+        response = []
+        return self.assertResult(
+            b"GET", b"/state/containers", None, OK, response
+        )
+
+    def test_one_container(self):
+        """
+        When the cluster state includes one container, the endpoint
+        returns a single-element list containing the container.
+        """
+        expected_application = Application(
+            name=u"myapp", image=DockerImage.from_string(u"busybox"))
+        expected_hostname = u"192.0.2.101"
+        self.cluster_state_service.update_node_state(
+            NodeState(
+                hostname=expected_hostname,
+                applications={expected_application},
+            )
+        )
+        expected_dict = dict(
+            name=u"myapp",
+            host=expected_hostname,
+            image=u"busybox:latest",
+            running=True,
+            restart_policy={u"name": u"never"},
+        )
```

Comment:
There's more fields that could possibly be returned in the response.  Pretty sure that will work since you re-used `container_configuration_response` but it might not be bad to have a test exercise them.